sklearn/datasets/_base.py

```diff
@@ -1119,16 +1143,17 @@ def load_sample_images():
     # import PIL only when needed
     from ..externals._pilutil import imread
 
-    module_path = join(dirname(__file__), "images")
-    with open(join(module_path, "README.txt")) as f:
-        descr = f.read()
-    filenames = [
-        join(module_path, filename)
-        for filename in sorted(os.listdir(module_path))
-        if filename.endswith(".jpg")
-    ]
-    # Load image data for each image in the source folder.
-    images = [imread(filename) for filename in filenames]
+    descr = load_descr(IMAGES_MODULE, 'README.txt')
+
+    images_module_traversable = importlib_resources.files(IMAGES_MODULE)
+    with importlib_resources.as_file(images_module_traversable) as images_dir:
```

Comment:
I think this is only available `python>=3.9`, so it could be problematic. So we probably need to find another (hopefully clean) way to do this.

Reply:
We can use the backport library `importlib_resources` for `python<=3.8` as noted [here](https://importlib-resources.readthedocs.io/en/latest/).

Reply:
@jackzyliu Do you know if this is the reason for the failures in the PR? If so shall we try to make sure everything else is working and leave the images for after?

Reply:
@jackzyliu Do you know if this is the reason for the failures in the PR? If so shall we try to make sure everything else is working and leave the images for after?

Reply:
Yea, it is. 

I will make the images portion backward-compatible.